src/main/scala/cromwell/util/TryUtil.scala

```diff
@@ -76,10 +76,35 @@ object TryUtil extends LazyLogging {
     }
   }
 
-  def sequence[A](s: Seq[Try[A]]): Try[Seq[A]] = {
-    s.collect({case f: Failure[_] => f}) match {
-      case failures if failures.nonEmpty => Failure(new AggregatedException(failures))
-      case _ => Success(s.map(_.get))
+  def sequence[A](s: Seq[Try[A]]): Try[Seq[A]] = flatten(s, "")
+
+  implicit class EnhancedTry[T](val t: Try[T]) extends AnyVal {
+    def getOrFail = t match {
+      case Success(s) => s
+      case Failure(f) => throw f
+    }
+  }
+
+  def flatten[T](tries: Seq[Try[T]], prefixErrorMessage: String): Try[Seq[T]] = {
+    tries collect { case f: Failure[_] => f } match {
+      case Nil => Success(tries map { _.get })
+      case failedTries => Failure(new AggregatedException(failedTries))
+    }
+  }
+
+  def flatten[T, U](tries: Map[T, Try[U]], prefixErrorMessage: String): Try[Map[T, U]] = {
+    tries collect { case (k, v: Failure[_]) => v } match {
```

Comment:
`k` isn't used below, if the failures are empty there's just a re-traversal of the map with `tries mapValues { _.get }`.

If you don't want to traverse the map or its values twice, maybe `partition` using an `.isSuccess` filter might be the way to go.